operator/pkg/reconciliation/reconcile_services.go

```diff
@@ -90,19 +90,23 @@ func (rc *ReconciliationContext) CheckHeadlessServices() result.ReconcileResult
 			return result.Error(err)
 
 		} else {
-			// if we found the service already, check if the labels are right
-			currentLabels := currentService.GetLabels()
-			desiredLabels := desiredSvc.GetLabels()
-			shouldUpdateLabels := !reflect.DeepEqual(currentLabels, desiredLabels)
-			if shouldUpdateLabels {
-				logger.Info("Updating labels",
+			// if we found the service already, check if they need updating
```

Comment:
getting the services updating in a proper/idiomatic-for-us way was a great side benefit of this work, great job!